File: app/src/main/java/com/venomvendor/sms/deduplicate/data/DuplicateList.java
package com.venomvendor.sms.deduplicate.data;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class DuplicateList<T> extends ArrayList<T> {

    public DuplicateList(Collection<? extends T> collection) {
        super(collection);
    }

    public List<T> splice(int fromIndex, int toIndex) {
        DuplicateList<T> splicedList = new DuplicateList<>(subList(fromIndex, toIndex));
        removeRange(fromIndex, splicedList.size());
        return splicedList;
    }
}

File: deduplicate-app/src/main/java/com/venomvendor/sms/deduplicate/data/DuplicateList.kt
package com.venomvendor.sms.deduplicate.data

import java.util.ArrayList

class DuplicateList<T>(collection: Collection<T>) : ArrayList<T>(collection) {

    fun splice(fromIndex: Int, toIndex: Int): List<T> {
        val splicedList: List<T> = DuplicateList(subList(fromIndex, toIndex))
        removeRange(fromIndex, splicedList.size)
        return splicedList
    }
}
